Sa 102293 kicker dod notification

bradbergeron-us opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary
Adds a new notification if kicker data is detected so the applicant can display an alert and prefill the question. Additionally, it removes some questions from the form.

## Related issue(s)
<img width="1017" alt="Screenshot 2025-01-07 at 10 11 19 AM" src="https://github.com/user-attachments/assets/9b45557a-c9ef-4230-80e5-9b2820577cae" />

<img width="830" alt="Screenshot 2025-01-07 at 10 10 55 AM" src="https://github.com/user-attachments/assets/d1dc701a-cbc3-4bf2-9ed9-af5b8789a1ea" />

## Testing done
Manually tested all conditions logged in as a new user with the feature flag ON and OFF.

## Screenshots
![Kicker_notification_active_duty](https://github.com/user-attachments/assets/fe8b8960-837f-4669-9153-93ba43e32e5e)
![kicker_notification_reserve](https://github.com/user-attachments/assets/11d3b3b4-e8c7-490a-9bdf-32e92eb845c5)

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?
My Education Benefits 1990 Application

## Acceptance criteria

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [X] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

bradbergeron-us committed:
Adds mebKickerNotificationEnabled to Original Claims form and copies it to formData

bradbergeron-us committed:
Refactors additional COnsiderations component to not use the old feature flags

bradbergeron-us committed:
Hides questions with new kicker notification turned On and displays alert if eligibility is detected

bradbergeron-us committed:
Implement kicker widget to display notification based on API data

bradbergeron-us committed:
Adds fallback for accessing kicker fields from claimant object

bradbergeron-us committed:
Adds try catch so that when properties are undefined on introduction page the application can still excecute until the values are fetched on the next page

bradbergeron-us committed:
Fix validation issue by treating street2 as a string, handling null/undefined values

bradbergeron-us committed:
Remove unused prefill transformer functions

bradbergeron-us committed:
Fix race condition with feature flags: only prefill kicker data when mebKickerNotificationEnabled is true

bradbergeron-us committed:
Adds conditional logic for preventing fetching of exclusion periods when mebKickerNotificationEnabled is true

bradbergeron-us committed:
Merge branch 'main' into SA-102293-kicker-dod-notification

bradbergeron-us committed:
Prevent fetching exclusion periods when mebKickerNotificationEnabled is true and remove mebKickerNotificationEnabled logic gate references

bradbergeron-us committed:
Fix redundant fetch of personal info when Chapter33 is selected

- Optimized useEffect to ensure personal info is only fetched once when a benefit is selected.
- Prevents duplicate calls to getPersonalInfo by checking if the chosen benefit has changed.
- Ensured previousChosenBenefit is updated after each fetch to avoid unnecessary API calls.

bradbergeron-us committed:
More refactoring

bradbergeron-us committed:
removes commenting out of moment removal debug

bradbergeron-us committed:
Fix: Safely handle falsy values for kicker eligibility

bradbergeron-us committed:
Adds custom notification message to submission if eligibility is detected

bradbergeron-us committed:
Updates abbreviation and capitalization of Defense

amasroor88 commented:
- Testing with feature flag enabled: [meb_kicker_notification_enabled]
<img width="717" alt="0128_local_ch1606_activeDuty_Yes" src="https://github.com/user-attachments/assets/eaeb8419-e1ef-4f52-82f9-a69302dacedc" />
<img width="1433" alt="0128_local_ch1606_reserveKickerData_Yes" src="https://github.com/user-attachments/assets/198c1585-62ae-4fec-ac65-1c308c52a5f9" />

bradbergeron-us committed:
Merge branch 'main' into SA-102293-kicker-dod-notification

bradbergeron-us committed:
updates prefill transformer to include only one version of the function

bradbergeron-us committed:
feat(test): Improve prefillTransformer coverage for mebKickerNotificationEnabled and address fallback
	•	Add scenarios testing mebKickerNotificationEnabled both ON and OFF
	•	Verify eligibility flags for active/reserve kickers populate correctly
	•	Expand tests for domestic vs. international address fallback
	•	Ensure missing phone fields are handled gracefully

bradbergeron-us committed:
fix(tests): Ensure sixHundredDollarBuyUp key is present to pass exclusion message test
	•	Added sixHundredDollarBuyUp: undefined to the test data for createAdditionalConsiderations
	•	Allows the function to produce 'N/A' for buy-up, matching the test’s expected output
	•	Fixes the failing unit test in form-submit-transform.unit.spec.js
